Modules/Core/ImageFunction/include/itkBSplineInterpolateImageFunction.hxx

```diff
@@ -58,10 +59,58 @@ void
 BSplineInterpolateImageFunction<TImageType, TCoordRep, TCoefficientType>::PrintSelf(std::ostream & os,
                                                                                     Indent         indent) const
 {
+  using namespace print_helper;
+
   Superclass::PrintSelf(os, indent);
-  os << indent << "Spline Order: " << m_SplineOrder << std::endl;
-  os << indent << "UseImageDirection = " << (this->m_UseImageDirection ? "On" : "Off") << std::endl;
-  os << indent << "NumberOfWorkUnits: " << m_NumberOfWorkUnits << std::endl;
+
+  os << indent << "Scratch: " << m_Scratch << std::endl;
+  os << indent
+     << "DataLength: " << static_cast<typename NumericTraits<typename TImageType::SizeType>::PrintType>(m_DataLength)
+     << std::endl;
+  os << indent << "SplineOrder: " << m_SplineOrder << std::endl;
+
+  itkPrintSelfObjectMacro(Coefficients);
+
+  os << indent << "MaxNumberInterpolationPoints: " << m_MaxNumberInterpolationPoints << std::endl;
+  os << indent << "PointsToIndex: " << m_PointsToIndex << std::endl;
+
+  itkPrintSelfObjectMacro(CoefficientFilter);
+
+  os << indent << "UseImageDirection: " << (m_UseImageDirection ? "On" : "Off") << std::endl;
+
+  os << indent
+     << "NumberOfWorkUnits: " << static_cast<typename NumericTraits<ThreadIdType>::PrintType>(m_NumberOfWorkUnits)
+     << std::endl;
+
+  os << indent << "ThreadedEvaluateIndex: ";
+  if (m_ThreadedEvaluateIndex != nullptr)
```

Comment:
Wasn't there a `PrintNullable` macro or something like that?

Reply:
Not that I know, and I have not been able to find it. If you happen to find it, I'd be happy to use it here. Otherwise, I maybe proposing it in a separate PR.

Reply:
What I had in mind is `itkPrintSelfObjectMacro`:
https://github.com/InsightSoftwareConsortium/ITK/blob/v5.3.0/Modules/Core/Common/include/itkMacro.h#L1263-L1275. I guess it is not applicable here?

Reply:
Only applies to pointers that can inherit from `itk::LightObject`.